### PR TITLE
allow selectors in noarch_platforms recipes

### PR DESCRIPTION
**Pull request by isuruf**
Checklist
* [x] Added a ``news`` entry

### CODE REVIEW

**Review by xhochy — APPROVED** [review]

File: conda_smithy/lint_recipe.py [hunk 1]
```diff
@@ -384,8 +384,27 @@ def lintify(meta, recipe_dir=None, conda_forge=False):
...
+            glob(os.path.join(recipe_dir, "..", "conda-forge.yml"))
+            or glob(
+                os.path.join(recipe_dir, "conda-forge.yml"),
+            )
+            or glob(
+                os.path.join(recipe_dir, "..", "..", "conda-forge.yml"),
+            )
+        )
+        if forge_yaml_filename:
+            with open(forge_yaml_filename[0], "r") as fh:
```
Comment (jaimergp) [inline conda_smithy/lint_recipe.py:398]: Shouldn't we check for `noarch_platforms` in all existing `conda-forge.yml`s, not just the first one we found? Or is it guaranteed that only one exists?

Comment (isuruf) [reply conda_smithy/lint_recipe.py:398]: There's only one used.

**Review by beckermr — APPROVED** [review]

**Comment by isuruf** [comment]
@jaimergp, ping

**Review by jaimergp — APPROVED** [review]

**Comment by BastianZim** [comment]
I’m still getting the lintong error: https://github.com/conda-forge/spyder-kernels-feedstock/pull/86#issuecomment-1306269223

Did I do something wrong?

File: conda_smithy/lint_recipe.py [hunk 2]
```diff
@@ -913,8 +941,10 @@ def is_selector_line(line):
         return False
     m = sel_pat.match(line)
     if m:
-        m.group(3)
-        return True
+        if allow_platforms and m.group(3) in ["win", "linux", "osx", "unix"]:
```
Comment (BastianZim) [inline conda_smithy/lint_recipe.py:944]: Shouldn't this also be the opposite? Meaning _not win_ etc.?

Comment (BastianZim) [reply conda_smithy/lint_recipe.py:944]: Thanks!

**Pull request merged**